### PR TITLE
LPS-121149 merge evaluation results with last edited values

### PR DESCRIPTION
**Pull request by linolaoj**
Related Issue: https://issues.liferay.com/browse/LPS-121149

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hey @linolaoj, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/thunks/fieldChange.es.js [hunk 1]
```diff
@@ -117,13 +121,20 @@ export default function fieldChange({
 						if (fieldInstance.isDisposed()) {
 							return;
 						}
+						const mergedPages = mergePages(
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/thunks/fieldChange.es.js:124]: @rijema, please add a break line here. Sorry for not let you know about this on my previous review. Usually, we separate variable declarations from statements, like `if`.

**Review by natocesarrego — COMMENTED** [review]
Hi @rijema and @linolaoj, I just have one last question regarding this pull, have you guys considered use the existent Array `REVALIDATE_UPDATES` to address the issue you're trying to solve with this proposed fix?

Basically, it collects the fields' changes performed by the user while an evaluation request is being processed on back-end and this Array is visited after the evaluation ends to add its changes to the `evaluatedPages` through the method `getEditedPages`.

Thanks.

**Comment by linolaoj** [comment]
Hi @natocesarrego,
I understand REVALIDATE_UPDATES was created to handle the operations on fields that are not evaluable. If we use it with evaluable fields, we could not reset the array on every evaluation like what is done in result https://github.com/liferay/liferay-portal/blob/c3ea0eda553683adf6aaa5a6d3afc21424e10a43/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/thunks/fieldChange.es.js#L108 and it could become a large array.

**Comment by natocesarrego** [comment]
Makes sense to me, @linolaoj. Thanks for your clarification. 😉 

Now, this pull just need to address this comment and then we're good to go. 🙂

**Comment by linolaoj** [comment]
Updated at https://github.com/liferay-forms/liferay-portal/pull/14

**Pull request closed without merging**